stl/inc/type_traits

```diff
@@ -521,6 +410,43 @@ _INLINE_VAR constexpr bool is_volatile_v<volatile _Ty> = true;
 template <class _Ty>
 struct is_volatile : bool_constant<is_volatile_v<_Ty>> {};
 
+// STRUCT TEMPLATE is_function
+template <class _Ty>
+_INLINE_VAR constexpr bool is_function_v =
+    !is_const_v<const _Ty> && !is_reference_v<_Ty>; // determine whether _Ty is a function
+
+template <class _Ty>
+struct is_function : bool_constant<is_function_v<_Ty>> {};
+
+// STRUCT TEMPLATE is_object
+template <class _Ty>
+_INLINE_VAR constexpr bool is_object_v = !is_function_v<_Ty> && !is_reference_v<_Ty> && !is_void_v<_Ty>;
+
+template <class _Ty>
+struct is_object : bool_constant<is_object_v<_Ty>> {}; // determine whether _Ty is an object type
+
+// STRUCT TEMPLATE is_member_function_pointer
+template <class>
+_INLINE_VAR constexpr bool is_member_function_pointer_v = // determine whether _Ty is a pointer to member function

```

Comment:
this comment is not only redundant but misleading, because the actual determining is done below!

I to wish that the answer to "is this a member function pointer" were always no, but alas it is sometimes yes and we must look below to figure out when